simulator.py

```diff
@@ -0,0 +1,62 @@
+# Class - Simulator
+# Runs multiple simulations of the game and saves the results
+
+import game
+import multiprocessing as mp
+import csv
+
+class Simulator:
+    """Class that manages simulations of a BlackJack game
+    
+    Parameters:
+
+    sim_count (int): Indicated the number of simulations to compute
+
+    """
+
+    def __init__(self, sim_count=100, bet_amount=1):
+        self.sim_count = sim_count
+        self.bet_amount = bet_amount
+
+    def main(self):
+        print("Starting Blackjack Simulator")
+        results = []
+        
+        #TODO: Implement concurrency
+        for x in range(self.sim_count):
+            results.append(self.run_game())
+        
+        self.write_results(results)
+        
+        print(f'Result for this simulation: {sum(results)/self.sim_count}')
+        print(f'Adjusted overall result: {self.get_overall_average(results)}')
+
+        # for x in range(10):
+        #     p = Process(target=run_game)
+        #     p.start()
+        #     p.join()
+
+
+    def run_game(self):
+        bj = game.Game()
+        result_dict = bj.game_result(bet=self.bet_amount)
+        return result_dict["result"]
+
+    def write_results(self, results):
+        res = open("results.csv", "a")
+        res.write(f'{self.sim_count},{self.bet_amount},{sum(results)}\n')
+        res.close()
+    
+    def get_overall_average(self, results):
```

Comment:
The more results we have the more work our program will have to do. I was thinking of having a separate file for the most recent overall results that would contain just the values for `total_sims` and `result_sum` - that way we could just increment those values and calculate the new average. We can continue to store the results in the csv for posterity.

Reply:
Good thinking I like that!

Reply:
Shweet, added!